components/clp-package-utils/clp_package_utils/scripts/native/archive_manager.py

```diff
@@ -0,0 +1,365 @@
+            db_conn.cursor(dictionary=True)
+        ) as db_cursor:
+
+            params = (begin_ts,)
+            query = f"SELECT id FROM `{table_prefix}archives` WHERE begin_timestamp >= %s"
+            if end_ts is not None:
+                query += " AND end_timestamp <= %s"
+                params = (begin_ts, end_ts)
+
+            db_cursor.execute(query, params)
+            results = db_cursor.fetchall()
+
+            archive_ids = [result["id"] for result in results]
+
+            if 0 == len(archive_ids):
+                logger.info("No archives found within specified time range.")
+                return 0
+
+            logger.info(f"Found {len(archive_ids)} archives within specified time range.")
+            for archive_id in archive_ids:
+                logger.info(archive_id)
+
+    except Exception:
+        logger.exception("Failed to find archives from the database.")
+        return -1
+
+    logger.info(f"Finished finding archives from the database.")
+
+    return 0
+
+
+def _delete_archives(
+    archives_dir: Path,
+    database_config: Database,
+    query: str,
+    params: List[str],
+    criteria: str,
+    dry_run: bool = False,
+) -> int:
+    """
+    Deletes archives from both metadata database and disk based on provided SQL query.
+    :param archives_dir:
+    :param database_config:
+    :param query: SQL query to select archives to delete.
+    :param params: List of parameters for SQL query.
+    :param criteria: Type of deletion criteria. Either "filter" or "ids".
+    :param dry_run: If True, no changes will be made to the database or disk.
+    :return: 0 on success, -1 otherwise.
+    """
+
+    archive_ids: List[str]
+    logger.info("Starting to delete archives from the database.")
+    try:
+        sql_adapter = SQL_Adapter(database_config)
+        clp_db_connection_params = database_config.get_clp_connection_params_and_type(True)
+        table_prefix = clp_db_connection_params["table_prefix"]
+        with closing(sql_adapter.create_connection(True)) as db_conn, closing(
+            db_conn.cursor(dictionary=True)
+        ) as db_cursor:
+
+            if dry_run:
+                logger.info("Running in dry-run mode. No changes will be made to the database.")
+
+            db_cursor.execute(query, params)
+            results = db_cursor.fetchall()
+
+            if 0 == len(results):
+                if "filter" == criteria:
+                    logger.info("No archives found within specified time frame.")
+                elif "ids" == criteria:
+                    logger.info("No archives found with matching IDs.")
+                return 0
+
+            archive_ids = [result["id"] for result in results]
+
+            if "ids" == criteria:
+                not_found_ids = set(params) - set(archive_ids)
+                if not_found_ids:
+                    logger.warning(
+                        f"""
+                        Failed to find archives with the following IDs:
```

Comment:
I am not sure if it should be "failed to delete" or "IDs don't exist".

I feel it should be latter?

```suggestion
                        Archives with the following IDs don't exist:
```